data/generate_test_data.py

```
@@ -0,0 +1,294 @@
+    
+    Notes:
+        On git command failure the function prints an error message and does not raise.
+    """
+    print(f"Pushing to {repo_url}...")
+    try:
+        subprocess.run(["git", "init"], cwd=target_dir, check=True)
+        subprocess.run(["git", "add", "."], cwd=target_dir, check=True)
+        subprocess.run(["git", "commit", "-m", "Initial commit with buggy code"], cwd=target_dir, check=True)
+        subprocess.run(["git", "branch", "-M", "main"], cwd=target_dir, check=True)
+        subprocess.run(["git", "remote", "add", "origin", repo_url], cwd=target_dir, check=True)
+        subprocess.run(["git", "push", "-u", "-f", "origin", "main"], cwd=target_dir, check=True)
```

Ruff (0.14.7)

Starting a process with a partial executable path

(S607)